node/src/services/blockchain/handler.rs

```diff
@@ -330,6 +335,42 @@ impl BlockchainService {
                             size,
                             user_peer_ids: peer_ids,
                         }),
+                        RuntimeEvent::FileSystem(
+                            pallet_file_system::Event::AcceptedBspVolunteer {
+                                who,
+                                location,
+                                fingerprint,
+                                multiaddresses,
+                                owner,
+                                size,
+                            },
+                        ) => {
+                            // We try to convert the types coming from the runtime into our expected types.
+                            let fingerprint: Fingerprint = fingerprint.as_bytes().into();
+                            // In this case, the Multiaddresses come as a BoundedVec of BoundedVecs of bytes,
+                            // and we need to convert them. It only fails if the bytes do not represent a valid multiaddress.
+                            // TODO: We should make sure Multiaddresses are valid from inside the runtime.
+                            let multiaddresses: Vec<Multiaddr> = multiaddresses
+                                .into_iter()
+                                .map({
+                                    |bounded_vec| {
+                                        bounded_vec
+                                            .into_inner()
+                                            .try_into()
+                                            .expect("Malformed Multiaddress.")
```

Comment:
Do not `.expect` here. If it fails, it will crash the blockchain service (kind of a big deal). Instead, just handle the error by ignoring the event. If the BSP volunteering does not have a valid multiaddress, it's his fault.

Reply:
You are right, we discussed this before. I'm now logging an error and just returning.